PyPDF2/_cmap.py

```diff
@@ -57,72 +122,104 @@ def parse_encoding(ft: DictionaryObject, space_code: int) -> Tuple[List[str], in
             if isinstance(o, int):
                 x = o
-            else:
+            else:  # isinstance(o,str):
                 try:
-                    encoding[x] = adobe_glyphs[o]
+                    encoding[x] = adobe_glyphs[o]  # type: ignore
                 except Exception:
-                    encoding[x] = o
+                    encoding[x] = o  # type: ignore
                     if o == " ":
                         space_code = x
                 x += 1
+    if isinstance(encoding, list):
+        encoding = dict(zip(range(256), encoding))
     return encoding, space_code
 
 
 def parse_to_unicode(ft: DictionaryObject, space_code: int) -> Tuple[Dict, int]:
-    map_dict: Dict[Any, Any] = {}
+    map_dict: Dict[
+        Any, Any
+    ] = (
+        {}
+    )  # will store all translation code and map_dict[-1] we will have the number of bytes to convert
     if "/ToUnicode" not in ft:
-        return map_dict, space_code
+        return {}, space_code
     process_rg: bool = False
     process_char: bool = False
-    cm: str = cast(DecodedStreamObject, ft["/ToUnicode"]).get_data().decode("utf-8")
-    for l in (
+    cm: bytes = cast(DecodedStreamObject, ft["/ToUnicode"]).get_data()
+    # we need to prepare cm before due to missing return line in pdf printed to pdf from word
+    cm = (
         cm.strip()
-        .replace("<", " ")
-        .replace(">", "")
-        .replace("[", " [ ")
-        .replace("]", " ] ")
-        .split("\n")
-    ):
-        if l == "":
+        .replace(b"beginbfchar", b"\nbeginbfchar\n")
+        .replace(b"endbfchar", b"\nendbfchar\n")
+        .replace(b"beginbfrange", b"\nbeginbfrange\n")
+        .replace(b"endbfrange", b"\nendbfrange\n")
+        .replace(b"<<", b"\n{\n")  # text between << and >> not used but
+        .replace(b">>", b"\n}\n")  # some solution to find it back
+    )
+    ll = cm.split(b"<")
+    for i in range(len(ll)):
+        j = ll[i].find(b">")
+        if j >= 0:
+            ll[i] = ll[i][:j].replace(b" ", b"") + b" " + ll[i][j + 1 :]
+    cm = (b" ".join(ll)).replace(b"[", b" [ ").replace(b"]", b" ]\n ")
+
+    for l in cm.split(b"\n"):
+        if l == b"" or l == b" ":
             continue
-        if "beginbfrange" in l:
+        if b"beginbfrange" in l:
             process_rg = True
-        elif "endbfrange" in l:
+        elif b"endbfrange" in l:
             process_rg = False
-        elif "beginbfchar" in l:
+        elif b"beginbfchar" in l:
             process_char = True
-        elif "endbfchar" in l:
+        elif b"endbfchar" in l:
             process_char = False
         elif process_rg:
-            lst = [x for x in l.split(" ") if x]
+            lst = [x for x in l.split(b" ") if x]
             a = int(lst[0], 16)
             b = int(lst[1], 16)
-            if lst[2] == "[":
+            nbi = len(lst[0])
+            map_dict[-1] = nbi // 2
+            fmt = b"%%0%dX" % nbi
+            if lst[2] == b"[":
```

Comment:
```suggestion
            if len(lst) < 3:
                continue
            if lst[2] == b"[":
```

Triggered by https://corpora.tika.apache.org/base/docs/govdocs1/982/982432.pdf

```
    return self._extract_text(self, self.pdf, space_width, PG.CONTENTS)
  File "/home/moose/Github/py-pdf/PyPDF2/PyPDF2/_page.py", line 1126, in _extract_text
    cmaps[f] = build_char_map(f, space_width, obj)
  File "/home/moose/Github/py-pdf/PyPDF2/PyPDF2/_cmap.py", line 20, in build_char_map
    map_dict, space_code = parse_to_unicode(ft, space_code)
  File "/home/moose/Github/py-pdf/PyPDF2/PyPDF2/_cmap.py", line 198, in parse_to_unicode
    if lst[2] == b"[":
IndexError: list index out of range
```

Reply:
no change required. previous corrections fixed the problem